test/api.jl

```diff
@@ -111,13 +167,8 @@ end
                     "results_format_version": "2.0.1",
                     "state": "COMPLETED"
                 }""")
-            @test rsp.metadata == [JSON3.read("""{
-                "relationId": "/:output/Int64",
-                    "types": [
-                                ":output",
-                                "Int64"
-                            ]
-            }""")]
+            # We unfortunately cannot test equality directly!
+            @test string(rsp.metadata) == string(make_proto_metadata())
```

Comment:
is this still true? i thought that with the new Protos package we _can_ test for equality now?

Reply:
Sadly, the default `==` comparison won't work if the proto contains e.g. arrays:
```julia
julia> struct MyStruct
         a::Int
         b::Vector{Int}
       end

julia> MyStruct(1, [2,3]) == MyStruct(1, [2,3])
false
```
If we want to, we could tweak the new codegen to optionally prepend everything with `@auto_hash_equals` or even generate the `==` and `hash` methods ourselves.